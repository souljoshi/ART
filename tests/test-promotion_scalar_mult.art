// check that scalar multiplication works with both
// double and int
// int is promoted to double

int main()
{
	int a = 10;
	double b = 5.5;
	vec v = <5.0, 5.0>;
	vec v2;
	vec v3;

	v2 = v * a; // int promoted to double
	v2 = a * v;

	v3 = b * v;
	v3 = v * b;

	printf(v2[0]);
	printc('\n');
	printf(v2[1]);
	printc('\n');
	printf(v3[0]);
	printc('\n');
	printf(v3[1]);
	printc('\n');
}

