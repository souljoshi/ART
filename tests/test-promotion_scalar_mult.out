50.000000
50.000000
27.500000
27.500000
